batchflow/models/tf/efficientnet.py

```diff
@@ -0,0 +1,318 @@
+                indicates whether the block can be scaled
+            other parameters :
+                see :func:`~.layers.conv_block` or base block documentation
+
+    head : dict
+        parameters for head
+
+    common : dict
+        common parameters
+        width_factor :
+        depth_factor : float
+            scaling factors
+    """
+
+    @classmethod
+    def default_config(cls):
+        config = super().default_config()
+
+        config['body'].update(dict(base=None,
+                                   blocks=[dict(repeats=1, scalable=True, layout='cna cna', kernel_size=3, filters=8)]))
+        config['head'].update(dict(layout='Pf'))
+
+        config['common/width_factor'] = 1.0
+        config['common/depth_factor'] = 1.0
+
+        config['loss'] = 'ce'
+        return config
+
+    def build_config(self, names=None):
+        config = super().build_config(names)
+
+        if config.get('head/units') is None:
+            config['head/units'] = self.num_classes('targets')
+
+        return config
+
+    @classmethod
+    def round_filters(cls, filters, factor):
+        """ Round number of filters based on width multiplier. """
+        return int(filters * factor)
+
+    @classmethod
+    def round_repeats(cls, repeats, factor):
+        """ Round number of layers based on depth multiplier. """
+        return int(np.ceil(repeats * factor))
+
+    @classmethod
+    def initial_block(cls, inputs, name='initial_block', **kwargs):
+        kwargs = cls.fill_params(name, **kwargs)
+        if kwargs.get('layout'):
+            return cls.block(inputs, name=name, **kwargs)
+        return inputs
+
+    @classmethod
+    def body(cls, inputs, name='body', **kwargs):
+        kwargs = cls.fill_params('body', **kwargs)
+
+        with tf.variable_scope(name):
+            blocks = kwargs.pop('blocks')
+            x = inputs
+            for i, block_args in enumerate(blocks):
+                x = cls.block(x, name='block-%d' % i, **{**kwargs, **block_args})
+
+            return x
+
+    @classmethod
+    def head(cls, inputs, name='head', **kwargs):
+        kwargs = cls.fill_params(name, **kwargs)
+        if kwargs.get('layout'):
+            return cls.block(inputs, name=name, **kwargs)
+        return inputs
+
+    @classmethod
+    def block(cls, inputs, name, **block_args):
+        """ block of repeated layers of same structure"""
+        base_block = block_args.pop('base', None)
+
+        new_layer_args = cls.update_layer_params(block_args)
+        if base_block is None:
+            with tf.variable_scope(name):
+                repeats = new_layer_args.get('repeats', 1)
+                x = inputs
+                for i in range(repeats):
+                    x = conv_block(x, name='layer-%d' % i, **new_layer_args)
+
+                return x
+        else:
+            return base_block(inputs, name=name, **new_layer_args)
+
+    @classmethod
+    def update_layer_params(cls, kwargs):
```

Comment:
Make so that function and `round_filters`/`round_repeates` are in the same location of the code: that would increase readability